### PR TITLE
Update package version

### PR DESCRIPTION
**Pull request by lyndaidaii**
Update newtonsoft.json/Servercommon/NuGet.Client SDK

Address: [#4452](https://github.com/NuGet/Engineering/issues/4452)

### CODE REVIEW

File: src/NuGet.Services.EndToEnd/Support/AssemblyMetadataPackageFile.cs [hunk 1]
```diff
@@ -28,6 +29,8 @@ public AssemblyMetadataPackageFile(string resourceName)
 
         public FrameworkName TargetFramework { get; set; }
 
+        public NuGetFramework NuGetFramework { get; set; }
```
Comment (shishirx34) [inline src/NuGet.Services.EndToEnd/Support/AssemblyMetadataPackageFile.cs:32]: why is this needed?

File: src/NuGet.Services.EndToEnd/NuGet.Services.EndToEnd.csproj [hunk 2]
```diff
@@ -157,11 +157,14 @@
     <PackageReference Include="Microsoft.Extensions.Configuration.Binder">
       <Version>2.2.0</Version>
     </PackageReference>
+    <PackageReference Include="Newtonsoft.Json">
+      <Version>13.0.1</Version>
+    </PackageReference>
```
Comment (shishirx34) [inline src/NuGet.Services.EndToEnd/NuGet.Services.EndToEnd.csproj:162]: since you are adding a top level dependency, doesn't this need binding redirect for transitive dependencies of Newtosoft.Json?

Comment (agr) [reply src/NuGet.Services.EndToEnd/NuGet.Services.EndToEnd.csproj:162]: I think we only need explicit binding redirects only for web projects (that use `web.config` instead of `app.config`).

Comment (joelverhagen) [reply src/NuGet.Services.EndToEnd/NuGet.Services.EndToEnd.csproj:162]: That's correct.

File: src/NuGet.Services.EndToEnd/Support/AssemblyMetadataPackageFile.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGet.Services.EndToEnd/Support/AssemblyMetadataPackageFile.cs:32]: The `IPackageFile` interface changed via the NuGet client package update.

**Review by joelverhagen — APPROVED** [review]
